core/src/test/java/com/linecorp/armeria/common/stream/ByteStreamMessageOutputStreamTest.java

```diff
@@ -0,0 +1,317 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.awaitility.Awaitility.await;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+
+import com.linecorp.armeria.common.HttpData;
+
+import reactor.test.StepVerifier;
+
+class ByteStreamMessageOutputStreamTest {
+
+    @Test
+    void write() {
+        final ByteStreamMessage byteStreamMessage = StreamMessage.fromOutputStream(os -> {
+            try {
+                for (byte b : "abcde".getBytes()) {
+                    os.write(b);
+                }
+                os.close();
```

Comment:
✅ updated, https://github.com/line/armeria/pull/4402/commits/267a6cd48bc47cf90a11353d935beec8d53111fd

Reply:
https://www.baeldung.com/java-try-with-resources was introduced in Java 7 which needs an explicit local variable.
Since Java 9, a local variable is not necessary for existing values. https://docs.oracle.com/en/java/javase/13/language/try-resources.html

Reply:
> Java 7, which needs an explicit local variable.
> Since Java 9, a local variable is not necessary for existing values.

aha~ I didn't know this! thanks for kind explanation 🙇